legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-dbExtension/legend-engine-xt-relationalStore-duckdb/legend-engine-xt-relationalStore-duckdb-pure/src/main/resources/core_relational_duckdb/relational/sqlQueryToString/duckdbExtension.pure

```diff
@@ -16,35 +30,183 @@ function <<db.ExtensionLoader>> meta::relational::functions::sqlQueryToString::d
+                dropTable = translateDropTableStatementDefault_DropTableSQL_1__DbConfig_1__String_1_,
+                loadTable = loadValuesToDbTableDefault_LoadTableSQL_1__DbConfig_1__String_MANY_
+              );
+}
+
 function <<access.private>> meta::relational::functions::sqlQueryToString::duckDB::getLiteralProcessorsForDuckDB():Map<Type,LiteralProcessor>[1]
 {
+  let literalReplacementPairForString = ^LiteralReplacement(old = '\'', new = '\'\'');
+
    newMap([
-   ]->cast(@Pair<Type,LiteralProcessor>))
+      
+      pair(StrictDate,     ^LiteralProcessor(format = '%s', transform = {d:StrictDate[1], dbTimeZone:String[0..1] | $d->convertDateToSqlStringDuckDB($dbTimeZone)})),
+      pair(DateTime,       ^LiteralProcessor(format = '%s', transform = {d:DateTime[1], dbTimeZone:String[0..1] | $d->convertDateToSqlStringDuckDB($dbTimeZone)})),
+      pair(Date,           ^LiteralProcessor(format = '%s', transform = {d:Date[1], dbTimeZone:String[0..1] | $d->convertDateToSqlStringDuckDB($dbTimeZone)}))
+   ]);
 }
 
+function meta::relational::functions::sqlQueryToString::duckDB::convertDateToSqlStringDuckDB(date:Date[1], dbTimeZone:String[0..1]):String[1]
+{
+   //Default to UTC, if timezone is not specified. GMT is the same as UTC, UTC is not actually a timezone
+   let timeZone = if( $dbTimeZone->isEmpty(), | 'GMT', |  $dbTimeZone->toOne());
+   if($date->hasSecond(),
+      | if ($date->hasSubsecond(),
+            | let d= format('%t{[' + $timeZone + ']yyyy-MM-dd HH:mm:ss.SSSSSS}', $date);
+              format('TIMESTAMP \'%s\'',$d);,
+            | let d= format('%t{[' + $timeZone + ']yyyy-MM-dd HH:mm:ss}', $date); 
+              format('TIMESTAMP_S \'%s\'',$d);
+           ),
+      | let d =format('%t{[' + $timeZone + ']yyyy-MM-dd}', $date);
+        format('DATE \'%s\'',$d);
+    );
+}
 function <<access.private>> meta::relational::functions::sqlQueryToString::duckDB::getDynaFunctionToSqlForDuckDB(): DynaFunctionToSql[*]
 {
   let allStates = allGenerationStates();
+  
   [
+    dynaFnToSql('adjust',                 $allStates,            ^ToSql(format='date_add(%s)', transform={p:String[3] | $p->at(0) + ',' + constructIntervalFunction($p->at(2), $p->at(1)) })),
+    dynaFnToSql('booland',                $allStates,            ^ToSql(format='every(%s)')),
+    dynaFnToSql('boolor',                 $allStates,            ^ToSql(format='any(%s)')),  
+    dynaFnToSql('castBoolean',            $allStates,            ^ToSql(format='cast(%s as boolean)')),  
+    dynaFnToSql('chr',                    $allStates,            ^ToSql(format='char(%s)')),
+    dynaFnToSql('concat',                 $allStates,            ^ToSql(format='concat%s', transform={p:String[*]|$p->joinStrings('(', ', ', ')')})),
+    // dynaFnToSql('convertDate',            $allStates,            ^ToSql(format='%s', transform={p:String[*] | $p->convertToDateH2()})),
```

Comment:
Can we remove these commented out lines that reference the H2 translation?

Reply:
just keeping those for reference these need to be implemented in next iteration